### ISSUE TITLE
server: no way to gracefully shut down connection

### ISSUE THREAD

**Issue by reynir** [mirage/awa-ssh#59]

**Comment by hannesm** [comment]
This doesn't fit your demands? https://github.com/mirage/awa-ssh/pull/52

**Comment by reynir** [comment]
No, it closes the connection somewhat abruptly (as noted in PR). I'd like to be able to send a message such as "Disconnecting you. Goodbye!" and have that arrive to the client before shutting down.

**Comment by dinosaure** [comment]
You probably can do something into the `Lwt_switch` hook.

**Comment by hannesm** [comment]
With #83, this is now done!?

**Comment by reynir** [comment]
Yes! I believe so.
